flips/20210118-self-ejection.md

```diff
@@ -0,0 +1,208 @@
+The established process for handling this case is (1) to revoke the compromised key and 
+(2) issue new key(s). Changing a node's keys mid-epoch will _not_ be supported for the foreseeable future.
+Therefore, the only available mitigation strategy is to remove the node as an authorized protocol participant 
+(this is essentially just 1). 
+
+Flow's technical term for revoking a node's participation privileges is `ejection` (from the protocol).
+This FLIP proposes `self-ejection` as a measure a node operator can take in case
+they believe their node's staking key has been compromised. Conceptually, self-ejection is very similar 
+to ejecting a node as punishment for protocol violations (usually accompanied by slashing the node's stake).
+
+
+## Motivation
+
+In case a node's staking key has been compromised, the attacker might be able to spoof node messages.
+For example, an attacker gaining knowledge of a consensus node's private staking and networking key 
+could impersonate the node and equivocate (double-vote or propose conflicting blocks). As a consequence,
+the node's stake will be slashed, and the node ejected. 
+
+For safety of the network, the operational rewards for a node are generally significantly smaller compared to
+a node's stake. Therefore, for an honest node operator, it is economically rational to forgo the 
+operational rewards to protect the stake from being slashed.
+
+
+## Benefits
+
+Supporting `self-ejection` benefits the node operators as well as the safety of the core protocol:
+* `self-ejection` allows the node operator to prevent significant economic losses through slashing
+* `self-ejection` increases protocol safety as this might prevent a malicious party from entering the network 
+
+
+## Design Proposal
+
+The node operator wishing to self-eject her node can submit their request via a transaction to the appropriate service account smart contract.
+That contract can then validate the request and propagate the resulting identity table change to the consensus committee via a service event.
+
+This matches the flow of information we have for identity table changes during epoch transitions:
+nodes submit staking requests to a smart contract, the requests are validated and aggregated into an identity table change,
+then the identity table change is propagated to the consensus committee (and ultimately to the protocol state) via a service event.
+
+![Illustration of the self-ejection process. For simplicity, we denote denote the node as well as the node operator requesting self-ejection as _Alice_ ](20210118-self-ejection/Self-Ejection.png)
+
+1. To request self-ejection, the operator (denoted as _Alice_) submits the respective transaction 
+   (denoted as _T_) to the network (same process as any other conventional transaction).
+   For security purposes, we might want to require that the request is signed by the staking key of the node that is requesting its own ejection.
```

Comment:
This transaction will require access to the resource obtained by the operator during staking registration. For additional security we may want to require that the request is signed by the staking key of the node requesting self-ejection.

Reply:
much better wording. Included in https://github.com/onflow/flow/pull/324/commits/f4e178e46f2cdfbcc464a444dd686eadac71734f